cpp/src/arrow/compute/kernels/scalar_cast_test.cc

```diff
@@ -1921,6 +1963,24 @@ TEST(Cast, BinaryOrStringToBinary) {
                 MaskArrayWithNullsAt(InvalidUtf8(to_type), {4}));
     }
   }
+
+  auto from_type = fixed_size_binary(3);
+  auto invalid_utf8 = FixedSizeInvalidUtf8(from_type);
+  CheckCast(invalid_utf8, invalid_utf8);
+  CheckCastFails(invalid_utf8, CastOptions::Safe(fixed_size_binary(5)));
```

Comment:
Or perhaps `CastOptions::Safe(from_type)`?

Reply:
That's covered by the CheckCast above; when casting fixed_size_binary to fixed_size_binary of the same width, then the cast succeeds. It's only casts between differing widths that fail.

Reply:
Ah, fair enough.